packages/expect/src/matchers.ts

```diff
@@ -134,16 +134,41 @@ const matchers: MatchersObject = {
     ensureNumbers(received, expected, matcherName, options);
 
     let pass = false;
-    let expectedDiff = 0;
-    let receivedDiff = 0;
-
-    if (received === Infinity && expected === Infinity) {
-      pass = true; // Infinity - Infinity is NaN
-    } else if (received === -Infinity && expected === -Infinity) {
-      pass = true; // -Infinity - -Infinity is NaN
+    let expectedDiff: number | bigint = 0;
+    let receivedDiff: number | bigint = 0;
+
+    //If both arent numbers, then one might be bigint
+    if (typeof received === 'number' && typeof expected === 'number') {
+      if (received === Infinity && expected === Infinity) {
+        pass = true; // Infinity - Infinity is NaN
+      } else if (received === -Infinity && expected === -Infinity) {
+        pass = true; // -Infinity - -Infinity is NaN
+      } else {
+        // Set default precision for numbers
+        precision = isNaN(precision) ? 2 : precision;
+        expectedDiff = Math.pow(10, -precision) / 2;
+        receivedDiff = Math.abs(expected - received);
+        pass = receivedDiff < expectedDiff;
+      }
     } else {
```

Comment:
I think we need parallel `} else if (typeof received === 'bigint' && typeof expected === 'bigint) {'

and then final `else { … }` for mixed number and bigint to throw matcher error

Reply:
[Commented the Test Example of having mixed](https://github.com/facebook/jest/pull/8382/files/fb6655d33f5efac890b77f199bd5b3f7684b1421#r279944898)

Doing the another `else if`, will remove the additional cast, but if we do allow mixed, we'll end up casting one of them.

Reply:
removed bigint support for tobeclosto